tests/optimism.rs

```diff
@@ -0,0 +1,125 @@
+#![cfg(feature = "optimism")]
+
+use alloy_primitives::Bloom;
+use alloy_provider::{Provider, ProviderBuilder};
+use alloy_rpc_types::{BlockId, BlockTransactionsKind};
+use op_alloy_network::Optimism;
+use pevm::{
+    chain::{PevmChain, PevmOptimism},
+    EvmAccount, Pevm, RpcStorage, Storage,
+};
+use reqwest::Url;
+use revm::primitives::SpecId;
+use tokio::runtime::Runtime;
+
+pub mod common;
+
+#[test]
+fn optimism_blocks_from_rpc() -> Result<(), Box<dyn std::error::Error>> {
+    let rpc_url = match std::env::var("OPTIMISM_RPC_URL") {
+        // The empty check is for GitHub Actions where the variable is set with an empty string when unset!?
+        Ok(value) if !value.is_empty() => value.parse()?,
+        _ => Url::parse("https://mainnet.optimism.io")?,
+    };
+
+    for block_number in [
+        // Here are random 8 blocks. The latter 6 have been commented because the CI is too slow.
+        111435579,
+        112003352,
+        // 114470424, 114673497, 118446712, 118931697, 121413120, 123129762,
+    ] {
+        dbg!(block_number);
+        let runtime = Runtime::new()?;
+        let provider = ProviderBuilder::<_, _, Optimism>::default().on_http(rpc_url.clone());
+        let block_opt = runtime.block_on(provider.get_block(
+            BlockId::number(block_number), //
+            BlockTransactionsKind::Full,
+        ))?;
+        let block = block_opt.ok_or("missing block")?;
+        let chain = PevmOptimism::mainnet();
+        let spec_id = chain.get_block_spec(&block.header).unwrap();
+        let pre_state_rpc_storage =
+            RpcStorage::new(provider.clone(), spec_id, BlockId::number(block_number - 1));
+
+        let concurrency_level =
+            std::thread::available_parallelism().unwrap_or(std::num::NonZeroUsize::MIN);
+
+        let mut pevm = Pevm::default();
+        let sequential_result = pevm.execute(
+            &pre_state_rpc_storage,
+            &chain,
+            block.clone(),
+            concurrency_level,
+            true,
+        );
+        let parallel_result = pevm.execute(
+            &pre_state_rpc_storage,
+            &chain,
+            block.clone(),
+            concurrency_level,
+            false,
+        );
+
+        assert_eq!(&sequential_result, &parallel_result);
```

Comment:
Can we share these checks with `common`/`mainnet`?

Reply:
I haven't found any meaningful abstraction yet.

The most reasonable plan to me is to change to:

```
assert_eq!(
            block.header.logs_bloom,
            PevmTxExecutionResult::get_logs_bloom(&tx_results)
);

assert_eq!(
            block.header.gas_used,
            PevmTxExecutionResult::get_cumulative_gas_used(&tx_results)
);
```

(or replace `PevmTxExecutionResult::` by `common::` if we don't want to pollute our lib).

------

Another way is to implement `assert_header(&block.header, &tx_results)`. This fn compares `block.header.receipts_root`, `block.header.logs_bloom`, `block.header.gas_used` to the `tx_results`. However, this fn name be misleading because we are only checking a subset of header.

In this case, I prefer WET (write everything twice) to DRY (don't repeat yourself), until we can find a meaningful abstraction.

Reply:
Being ad-hoc is fine for less important chains. Here OP (i.e., RISE as well) and Ethereum are of the same importance. It's weird to see one having on-disk checks while the others don't, then the other having `observed_accounts` and `expected_storage` checks that one doesn't, etc. The whole structure, flow, and what to check should be very similar. 

This is not trivial, so perhaps we can start with some mock blocks (like early day Ethereum), then complete OP tests in the upcoming PRs.